library/core/src/intrinsics.rs

```diff
@@ -2013,21 +2013,27 @@ extern "rust-intrinsic" {
     pub fn ptr_offset_from_unsigned<T>(ptr: *const T, base: *const T) -> usize;
 
     /// See documentation of `<*const T>::guaranteed_eq` for details.
+    /// Returns `2` if the result is unknown.
+    /// Returns `1` if the pointers are guaranteed equal
+    /// Returns `0` if the pointers are guaranteed inequal
     ///
     /// Note that, unlike most intrinsics, this is safe to call;
     /// it does not require an `unsafe` block.
     /// Therefore, implementations must not require the user to uphold
     /// any safety invariants.
+    ///
+    /// Do not call this at runtime, use `const_eval_select` to invoke regular
+    /// pointer arithmetic instead.
```

Comment:
Why this, given that it has a runtime implementation?

Reply:
leftover comment from a previous implementation attempt. I wanted to use const eval select to avoid the codegen impls, but that didn't work out